### PR TITLE
added isPublic flag

### PR DESCRIPTION
**Pull request by wotwot563**
## Description

The email will only display if IsPublic is set to true.
fixed bug with embedded role seeding on admin.

## Type of change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] I updated the changelog with an end-user readable description
-   [ ] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

run all the endpoints except the user endpoint with user in the return object.
if the userflag is public is false the email should be <Redacted>

## Link to issue

Closes: #138

### CODE REVIEW

File: Models/Defaults/Defaults.cs [hunk 1]
```diff
@@ -23,6 +23,13 @@ namespace Models.Defaults
     public static class Defaults
     {
 
+        public static class Privacy
```
Comment (StijnGroenen) [inline Models/Defaults/Defaults.cs:26]: Missing documentation

File: Models/User.cs [hunk 2]
```diff
@@ -49,6 +49,7 @@ public User()
 
         public string ProfileUrl { get; set; }
 
+        public bool IsPublic { get; set; }
```
Comment (StijnGroenen) [inline Models/User.cs:52]: Missing documentation

File: Repositories.Tests/EmbeddedProjectRepositoryTest.cs [hunk 3]
```diff
@@ -115,15 +116,131 @@ public async Task GetEmbeddedProject_goodflow([EmbeddedDataSource]EmbeddedProjec
...
         public async Task GetEmbeddedProjects_noProject([EmbeddedDataSource]EmbeddedProject project)
         {
             // Test
             Assert.IsNull( await Repository.GetEmbeddedProjectAsync(project.Guid));
         }
+
+        /// <summary>
+        /// Finds the asynchronous user is public true.
+        /// </summary>
+        /// <param name="project">The embeddedProject.</param>
```
Comment (StijnGroenen) [inline Repositories.Tests/EmbeddedProjectRepositoryTest.cs:132]: The documentation in this class (mostly the parameter documentation) is very vague. Please add a proper description of what the parameters do.

File: Repositories.Tests/ProjectRepositoryTest.cs [hunk 4]
```diff
@@ -75,9 +76,54 @@ public async Task GetAllWithUsersAsyncTest_NoProjects([UserDataSource(100)]List<
...
+            // Test
+            List<Project> retrieved = await Repository.GetAllWithUsersAsync();
+            Assert.AreEqual(1, retrieved.Count);
+            Assert.AreEqual(retrieved[0].User.Email, Defaults.Privacy.RedactedEmail);
+        }
+
+        /// <summary>
+        /// Gets all with users asynchronous test is public email false.
+        /// </summary>
+        /// <param name="project">The project.</param>
```
Comment (StijnGroenen) [inline Repositories.Tests/ProjectRepositoryTest.cs:103]: The documentation in this class (mostly the parameter documentation) is very vague. Please add a proper description of what the parameters do.

File: Repositories/EmbedRepository.cs [hunk 5]
```diff
@@ -60,31 +61,81 @@ public class EmbedRepository : Repository<EmbeddedProject>, IEmbedRepository
...
+            {
+                embeddedProjects[i] = RedactUser(embeddedProjects[i]);
+            }
+            return embeddedProjects;
+        }
+
+        /// <summary>
+        /// Finds the asynchronous.
+        /// </summary>
+        /// <param name="id">The identifier.</param>
```
Comment (StijnGroenen) [inline Repositories/EmbedRepository.cs:99]: The identifier of what?

The documentation in this class (mostly the parameter documentation) is very vague. Please add a proper description of what the parameters do.

File: Models/User.cs [hunk 6]
```diff
@@ -49,6 +49,13 @@ public User()
 
         public string ProfileUrl { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether the user profile is public.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if this instance is public; otherwise, <c>false</c>.
+        /// </value>
+        public bool IsPublic { get; set; }
```
Comment (Brend-Smits) [inline Models/User.cs:58]: We should set the default to false. 
We want this to be an opt-in feature.

Comment (wotwot563) [reply Models/User.cs:58]: The default value is false

Comment (Brend-Smits) [reply Models/User.cs:58]: Right, I keep forgetting this. My bad......

File: Models/Defaults/Defaults.cs (hunk 1 above)
Comment (StijnGroenen) [reply Models/Defaults/Defaults.cs:28]: What does this constant do?

File: Repositories.Tests/ProjectRepositoryTest.cs [hunk 7]
```diff
@@ -362,7 +408,75 @@ public async Task SearchCountAsyncTest_BadFlow_NoProjects()
...
+            Assert.AreEqual(project.Uri, retrieved.Uri);
+            Assert.AreEqual(project.User.Name, retrieved.User.Name);
+            Assert.AreEqual(project.User.Email, retrieved.User.Email);
+            Assert.AreEqual(project.Collaborators[0].FullName, retrieved.Collaborators[0].FullName);
+        }
+
+        /// <summary>
+        /// Finds the with user and collaborators asynchronous test is public true.
+        /// </summary>
+        /// <param name="project">The project.</param>
```
Comment (StijnGroenen) [inline Repositories.Tests/ProjectRepositoryTest.cs:447]: Missing clear parameter documentation.
What do these parameters do?

**Pull request merged**